easybuild/easyconfigs/j/jax/jax-0.2.18-foss-2020b.eb

```diff
@@ -0,0 +1,88 @@
+# This file is an EasyBuild reciPY as per https://github.com/easybuilders/easybuild
+# Author: Denis Kristak
+easyblock = 'PythonBundle'
+
+name = 'jax'
+version = '0.2.18'
+local_jaxlib_ver = '0.1.70'
+homepage = 'https://pypi.python.org/pypi/jax'
+description = """Composable transformations of Python+NumPy programs: differentiate, vectorize, JIT to GPU/TPU, and
+more"""
+
+toolchain = {'name': 'foss', 'version': '2020b'}
+
+# downloading TF tarball for better control of what Bazel will use during installation of Jax
+local_tf_tarball_version = '2.5.0'
+
+builddependencies = [
+    ('Bazel', '3.7.2'),
+]
+
+dependencies = [
+    ('Python', '3.8.6'),
+    ('SciPy-bundle', '2020.11'),
+    ('Cython', '0.29.22'),
+    ('flatbuffers-python', '1.12'),
+]
+
+local_tf_builddir = "%%(builddir)s/tensorflow-%s" % local_tf_tarball_version
+
+local_jax_prebuildopts = "sed -i -e 's$pathToSed$%s$g' WORKSPACE && " % local_tf_builddir
+
+local_jax_build_cmd = 'python build/build.py '
+local_jax_build_cmd += '--bazel_startup_options="--output_user_root=%(builddir)s" '
+local_jax_build_cmd += '--bazel_path="$EBROOTBAZEL/bin/bazel" '
+local_jax_build_cmd += '--bazel_options=--subcommands '
+local_jax_build_cmd += '--bazel_options=--jobs=1 --bazel_options=--action_env=PYTHONPATH '
```

Comment:
It would be really nice if we could use `--jobs=%(parallel)s` here to avoid building `jax` with just a single core, but it seems like this may be breaking the build...

Here's the build error I was hitting when using `--jobs=%(parallel)s`:

```
INFO: Build completed successfully, 5551 total actions
INFO: Running command line: bazel-bin/build/build_wheel '--output_path=/tmp/vsc40023/easybuild_build/jax/0.2.18/foss-2020b/jax-jaxlib-v0.1.70/dist' '--cpu=x86_64'
INFO: Build completed successfully, 5551 total actions
Traceback (most recent call last):
  File "/tmp/vsc40023/easybuild_build/jax/0.2.18/foss-2020b/134f18b726e1fd90f125772c3f16cba2/execroot/__main__/bazel-out/k8-opt/bin/build/build_wheel.runfiles/__main__/build/build_wheel.py", line 252, in <module>
    prepare_wheel(sources_path)
  File "/tmp/vsc40023/easybuild_build/jax/0.2.18/foss-2020b/134f18b726e1fd90f125772c3f16cba2/execroot/__main__/bazel-out/k8-opt/bin/build/build_wheel.runfiles/__main__/build/build_wheel.py", line 210, in prepare_wheel
    patch_copy_xla_extension_stubs(jaxlib_dir)
  File "/tmp/vsc40023/easybuild_build/jax/0.2.18/foss-2020b/134f18b726e1fd90f125772c3f16cba2/execroot/__main__/bazel-out/k8-opt/bin/build/build_wheel.runfiles/__main__/build/build_wheel.py", line 106, in patch_copy_xla_extension_stubs
    with open(r.Rlocation(
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

I'm testing again now (with the changes from https://github.com/deniskristak/easybuild-easyconfigs/pull/3 included) to confirm that problem is indeed due to the parallel build. If that's the case, I'll open an issue in the `jax` GitHub repo to ask whether building in parallel is supposed to be working...

Reply:
fixed in 1210ab0, helps a lot with speeding up the build :)